cli/graph_util.rs

```diff
@@ -97,6 +100,25 @@ pub fn graph_valid(
           roots.contains(error.specifier())
         }
       };
+
+      if is_root
+        && options.ignore_root_export_errors
+        && matches!(
+          &error,
+          ModuleGraphError::ModuleError(ModuleError::LoadingErr(
+            _,
+            _,
+            ModuleLoadError::Jsr(JsrLoadError::UnknownExport {
+              export_name,
+              ..
+            }),
+          )) if export_name == "."
+        )
+      {
+        log::debug!("Ignoring: {}", error);
+        return None;
+      }
```

Comment:
> This error arises when we attempt to cache packages listed in the import map after running deno add. The way we do this isn't particularly smart, we just attempt to cache everything that looks like a complete package path (no trailing slash).

Can't we instead get the exports for the package in pm/add.rs and only cache those?